src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.RuntimeIdentifierInference.targets

```diff
@@ -71,7 +107,8 @@ Copyright (c) .NET Foundation. All rights reserved.
                     '$(SelfContained)' == 'true' or
                     '$(PublishReadyToRun)' == 'true' or
                     '$(PublishSingleFile)' == 'true' or
-                    '$(PublishAot)' == 'true'
+                    '$(PublishAot)' == 'true' or
+                    ('$(_PublishSelfContainedIsActive)' == 'true' and '$(PublishSelfContained)' == 'true')
```

Comment:
Is this extra clause necessary?  It seems if these two properties are true then `SelfContained` would also be true, which is already covered earlier on.